server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -109,52 +108,36 @@ func (p *FetchServer) FetchBlob(ctx context.Context, req *rapb.FetchBlobRequest)
 		return nil, err
 	}
 
-	var expectedSHA256 string
-
+	storageFunc := req.GetDigestFunction()
+	if storageFunc == repb.DigestFunction_UNKNOWN {
+		storageFunc = repb.DigestFunction_SHA256
+	}
+	var checksumFunc repb.DigestFunction_Value
```

Comment:
it seems inconvenient for servers that the storage digest can differ from the checksum digest. Is it too late to change this design? If not, I wonder if it'd make sense to enforce that they are equal (by ignoring `--digest_function` just for remote asset fetches and instead just using whatever digest has been supplied in the fetch target)

Reply:
We could add comments into the remote-apis spec with something like "MUST be equal" or something similar?

Realistically, I have implemented it on the Bazel side so that the CRI checksum will be the same as the request's digest function. So we should only expect misuse from non-Bazel clients.

Reply:
I think adding the "MUST be equal" comment would be ideal. And if they're not equal we could return an error

Reply:
Raised https://github.com/bazelbuild/remote-apis/pull/294

It might take a while to get merged though. I think we should proceed with the current implementation and remove the redundant code once the API spec is updated. WDYT?

Reply:
sgtm